core/arch/arm/kernel/boot.c

```diff
@@ -1473,6 +1473,31 @@ struct ns_entry_context *boot_core_hpen(void)
 }
 #endif
 
+#if defined(CFG_CORE_STACK_PROTECTOR)
+/* generate random stack canary value on boot up */
+__weak unsigned long boot_get_random_stack_canary(void)
```

Comment:
Change here to `uintptr_t` for con, consistency?

If platform can legitimatly override this function, it should better be prefixed `plat_` (e.g. `plat_get_random_stack_canary()`).